### PR TITLE
Adds metrics initialization metrics for the service interface(#12850)

### PR DESCRIPTION
**Pull request by robin977**
Adds metrics initialization metrics for the service interface(#12850)

### CODE REVIEW

**Review by AlbumenJ — COMMENTED** [review]
@songxiaosheng @wxbty PTAL

**Comment by songxiaosheng** [comment]
init metric as follow
image

**Review by songxiaosheng — APPROVED** [review]
LGTM

File: dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/data/MethodStatComposite.java [hunk 1]
```diff
@@ -54,6 +55,13 @@ public void initWrapper(List<MetricsKeyWrapper> metricsKeyWrappers) {
         metricsKeyWrappers.forEach(appKey -> methodNumStats.put(appKey, new ConcurrentHashMap<>()));
     }
 
+    public void initMethodKey(MetricsKeyWrapper wrapper, Invocation invocation, int size) {
+        if (!methodNumStats.containsKey(wrapper)) {
+            return;
+        }
+        methodNumStats.get(wrapper).computeIfAbsent(new MethodMetric(getApplicationModel(), invocation), k -> new AtomicLong(0L)).set(size);
+    }
+
```
Comment (wxbty) [inline dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/data/MethodStatComposite.java:64]: The init method can directly set 0, no need to transfer parameters

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/AggregateMetricsCollector.java [hunk 2]
```diff
@@ -123,12 +124,56 @@ public boolean isSupport(MetricsEvent event) {
     public void onEvent(RequestEvent event) {
         if (enableQps) {
             MethodMetric metric = calcWindowCounter(event, MetricsKey.METRIC_REQUESTS);
+            if(event instanceof RequestInitEvent) {
+                onInitMetricRequest(event);
+                onInitQpsRequest(metric);
+                onInitRtRequest(metric);
+                onInitRtAgrRequest(metric);
+                return;
+            }
```
Comment (wxbty) [inline dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/AggregateMetricsCollector.java:133]: RequestInitEvent and RequestEvent look different, one is triggered by app init, the other is triggered by rpc request, is it better to separate the listener?

**Review by wxbty — CHANGES_REQUESTED** [review]

**Comment by robin977** [comment]
企业微信截图_705265f9-eee7-427e-8456-4cf3cffc1b99
企业微信截图_db714257-9123-4f99-bca5-ff472448b4a4
企业微信截图_340eefef-f167-4d51-b07e-846e75a78469

File: dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/event/MetricsInitEvent.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+
+    public MetricsInitEvent(ApplicationModel source,TypeWrapper typeWrapper) {
+        super(source,typeWrapper);
+    }
+
+    public static MetricsInitEvent toMetricsInitEvent(ApplicationModel applicationModel, Invocation invocation) {
+        MethodMetric methodMetric = new MethodMetric(applicationModel, invocation);
+        MetricsInitEvent initEvent = new MetricsInitEvent(applicationModel, new TypeWrapper(MetricsLevel.SERVICE, METRIC_REQUESTS));
+        initEvent.putAttachment(MetricsConstants.INVOCATION, invocation);
```
Comment (wxbty) [inline dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/event/MetricsInitEvent.java:38]: Do not create immutable objects every time.

**Review by wxbty — CHANGES_REQUESTED** [review]

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/event/DefaultSubDispatcher.java [hunk 4]
```diff
@@ -69,7 +70,7 @@ private CategoryOverall initMethodRequest() {
                             MetricsSupport.increment(key, dynamicPlaceType, (MethodMetricsCollector) collector, event);
                             MetricsSupport.increment(MetricsKey.METRIC_REQUESTS_PROCESSING, dynamicPlaceType, (MethodMetricsCollector) collector, event);
                         })),
-                new MetricsCat(MetricsKey.METRIC_REQUESTS_SUCCEED, (key, placeType, collector) -> AbstractMetricsKeyListener.onFinish(key,
+                new MetricsCat(METRIC_REQUESTS_SUCCEED, (key, placeType, collector) -> AbstractMetricsKeyListener.onFinish(key,
```
Comment (wxbty) [inline dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/event/DefaultSubDispatcher.java:73]: Don't change that. Keep it the same as before

**Review by wxbty — CHANGES_REQUESTED** [review]

**Comment by wxbty** [comment]
LGTM

**Review by wxbty — APPROVED** [review]

**Pull request merged**